can/interfaces/ixxat/canlib.py

```diff
@@ -482,7 +487,12 @@ def _recv_internal(self, timeout):
         return rx_msg, True
 
     def send(self, msg, timeout=None):
-
+        """
+        Sends a message on the bus. The interface may buffer the message.
+        returns True on success or when timeout is None
+        returns False on timeout (when timeout is not None)
+        raises CanOperationError
```

Comment:
Hmm. Thats a good point. Maybe our way thinking about Exceptions differs a bit. This method is a good example for "fine-adjusting" the handling/rasising of exceptions.

We have two cases: With or without timeout
First without timeout: Thats easy. The user don't want a timeout, so any timeout is a case that should not happen -> exceptional behavior which needs special treatment.
Second with timeout: Thats the point. The user wanted a timeout, so he/she accepts any timeout (maybe don't cares about it - recovering from a fault situation might be a situation were this mades much sense. the fault may still be present and the other node not reachable). So is an timeout an exceptional behavior? Not really. (But i still need to clearify if an timeout may occur because of exceptional behavior of the driver. Then this can get complicated.)

Reply:
In any case the behavior should be consistent between interfaces. If we change to return a boolean instead of raising an exception then other interfaces must change too.

My original intention was that if a timeout is specified, the developer really wants to make sure the message is sent or queued successfully. If not then something is wrong and the program may be stopped or it can try to retransmit later.

Reply:
Ok, I'll stick to the more pythonic raise-an-exception way.
BUT. Should we generally use the built-in TimeoutError for timeouts or sub-class CanError?
I prefer the built-in.

Reply:
Would there be an issue inheriting from both? I agree that our specific exception e.g. `CanBusWriteTimeout` should inherit from `TimeoutError` and I imagine it should work also inheriting from our generic `CanError` exception?

Reply:
I think there would be no problem. But i would call this class CanTimeoutError and use it generally for timeout errors, not only for send.

Reply:
On the other hand I don’t think the OSErrors are supposed to be used for other than pure system level errors.

“Raised when a system function timed out at the system level. Corresponds to errno ETIMEDOUT.”

Reply:
Oki. I had a look into other python packages and the common unterstanding is like christian said it. They dervie their own TimeoutError.
So we'll get an CanTimeoutError